Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/OAuthClientsRepository.cs

```diff
@@ -13,21 +14,38 @@ namespace Backbone.Modules.Devices.Infrastructure.Persistence.Repository;
 public class OAuthClientsRepository : IOAuthClientsRepository
 {
     private readonly OpenIddictApplicationManager<CustomOpenIddictEntityFrameworkCoreApplication> _applicationManager;
+    private List<CustomOpenIddictEntityFrameworkCoreApplication> _cachedApplications;
```

Comment:
You should use a dictionary instead, so that the read has a better performance. It's also easier to use.